### PR TITLE
[CDAP-7346] Add Application Version in ApplicationSpecification

### PR DESCRIPTION
**Pull request by ecfm**
https://issues.cask.co/browse/CDAP-7346
http://builds.cask.co/browse/CDAP-RUT191-5

### CODE REVIEW

**Comment by ecfm** [comment]
build is green

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/DefaultApplicationSpecification.java [hunk 1]
```diff
@@ -68,6 +70,38 @@ public DefaultApplicationSpecification(String name, String description, String c
                                          Map<String, WorkerSpecification> workers,
                                          Map<String, Plugin> plugins) {
     this.name = name;
+    this.appVersion = ApplicationId.DEFAULT_VERSION;
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/DefaultApplicationSpecification.java]: can we do just `this(name, ApplicationId.DEFAULT_VERSION, description, ......)`

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java [hunk 2]
```diff
@@ -87,6 +89,25 @@ public InMemoryConfigurator(CConfiguration cConf, Id.Namespace appNamespace, Id.
...
                                   cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
+    this.applicationVersion = ApplicationId.DEFAULT_VERSION;
+  }
+
+  public InMemoryConfigurator(CConfiguration cConf, Id.Namespace appNamespace, Id.Artifact artifactId,
+                              String appClassName, ArtifactRepository artifactRepository,
+                              ClassLoader artifactClassLoader,
+                              @Nullable String applicationName, @Nullable String applicationVersion,
+                              @Nullable String configString) {
+    this.cConf = cConf;
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java:91]: same comment as above

File: cdap-proto/src/main/java/co/cask/cdap/proto/ApplicationDetail.java [hunk 3]
```diff
@@ -50,7 +52,29 @@ public ApplicationDetail(String name,
                            List<PluginDetail> plugins,
                            ArtifactSummary artifact) {
     this.name = name;
-    this.version = artifact.getVersion();
+    this.appVersion = ApplicationId.DEFAULT_VERSION;
```
Comment (gokulavasan) [inline cdap-proto/src/main/java/co/cask/cdap/proto/ApplicationDetail.java]: same comment as above

**Comment by gokulavasan** [comment]
Few comments regarding the constructors. Rest looks good.

**Comment by ecfm** [comment]
build is green

**Comment by gokulavasan** [comment]
LGTM 👍

**Pull request merged**